### ISSUE TITLE
Static generator should honor `IGNORE_FILES`

### ISSUE THREAD

**Issue by zackw** [getpelican/pelican#1692]
Files that match an `IGNORE_FILES` glob should not be copied by the static generator.  This is not just cosmetic; emacs lock files (`'.#*'`, included in `IGNORE_FILES` by default) are pseudo-symlinks (that is, symlinks used as an atomic way to store a small amount of data in the file system -- the destination of the symlink never exists) and `shutil.copy2` will crash if it tries to copy them.

**Comment by avaris** [comment]
Static generator does honor `IGNORE_FILES`. If you're getting tracebacks, please add them.

**Comment by zackw** [comment]
@avaris I suppose it would be more precise to say 'the static generator doesn't apply `IGNORE_PATHS` to files nested inside a directory copied by `_copy_paths`', but I don't see why, with my user hat on, I should have to know that those are not treated the same as other static files.

Example traceback:

~~~~
--- Logging error ---
Traceback (most recent call last):
  File "/home/zack/owlfolio/pelican-core/pelican/__init__.py", line 437, in main
    pelican.run()
  File "/home/zack/owlfolio/pelican-core/pelican/__init__.py", line 177, in run
    p.generate_output(writer)
  File "/home/zack/owlfolio/pelican-core/pelican/generators.py", line 720, in generate_output
    os.curdir)
  File "/home/zack/owlfolio/pelican-core/pelican/generators.py", line 689, in _copy_paths
    os.path.join(output_path, destination, final_path))
  File "/home/zack/owlfolio/pelican-core/pelican/utils.py", line 310, in copy
    recurse(source_, destination_)
  File "/home/zack/owlfolio/pelican-core/pelican/utils.py", line 306, in recurse
    shutil.copy2(entry_path, destination)
  File "/home/zack/owlfolio/pelican-venv/lib/python3.4/shutil.py", line 245, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/home/zack/owlfolio/pelican-venv/lib/python3.4/shutil.py", line 108, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/home/zack/owlfolio/www/style/static/.#main.scss'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.4/logging/__init__.py", line 978, in emit
    msg = self.format(record)
  File "/usr/lib/python3.4/logging/__init__.py", line 828, in format
    return fmt.format(record)
  File "/home/zack/owlfolio/pelican-core/pelican/log.py", line 25, in format
    record.msg = record.msg.replace('\n', '\n  | ')
AttributeError: 'tuple' object has no attribute 'replace'
Call stack:
  File "/home/zack/owlfolio/pelican-venv/bin/pelican", line 9, in <module>
    load_entry_point('pelican==3.5.0', 'console_scripts', 'pelican')()
  File "/home/zack/owlfolio/pelican-core/pelican/__init__.py", line 447, in main
    logger.critical(e.args)
  File "/usr/lib/python3.4/logging/__init__.py", line 1322, in critical
    self._log(CRITICAL, msg, args, **kwargs)
  File "/home/zack/owlfolio/pelican-core/pelican/log.py", line 141, in _log
    exc_info=exc_info, extra=extra)
Message: (2, 'No such file or directory')
Arguments: ()
CRITICAL: [Errno 2] No such file or directory: '/home/zack/owlfolio/www/style/static/.#main.scss'
~~~~

**Comment by avaris** [comment]
As a user, you don't have to *know* what caused to problem, that's true. That's also why I asked for a traceback. Now it's easier to see that, Static generator doesn't check for ignore files when copying theme.

**Comment by avaris** [comment]
Conveniently, the traceback also points to another bug in logging. So, thank you.

**Comment by zackw** [comment]
I don't know why it's not showing up here, but pull request #1693 should fix this bug (but not the logging bug).

**Comment by justinmayer** [comment]
Ostensibly addressed by #1693.
